### PR TITLE
Add signin with email endpoint and tests

### PR DESCRIPTION
**Pull request by Aumarh**

### CODE REVIEW

File: backend/src/contexts/users/actions/SignInAction.ts [hunk 1]
```diff
@@ -18,4 +18,13 @@ export class SignInAction {
     if (match) return user;
     else throw new WrongPasswordException();
   }
+
+  async executeWithEmail(email: string, password: string) {
```
Comment (hynnot) [inline backend/src/contexts/users/actions/SignInAction.ts:22]: I think maybe it's interesting to put this behavoiur in the execute function, to don't duplicate a lot lines here and in the app file.

**Review by hynnot — CHANGES_REQUESTED** [review]

File: backend/src/contexts/users/actions/SignInAction.test.ts [hunk 2]
```diff
@@ -6,38 +6,48 @@ import { UserRepositoryInMemory } from '../infrastructure/persistence/repositori
...
+    userRepository.storeUser(new User(email, username, password));
     const signInAction = new SignInAction(userRepository);
 
     await expect(
-      signInAction.execute(username, 'wrong password'),
+      signInAction.execute('wrong password', username, email),
     ).rejects.toThrow(WrongPasswordException);
   });
 
   test('can sign in', async () => {
```
Comment (hynnot) [inline backend/src/contexts/users/actions/SignInAction.test.ts:40]: This test it's can sign in by username, but we need another about can sign in by email

File: backend/src/tests/e2e/signin.test.ts [hunk 3]
```diff
@@ -25,8 +25,9 @@ describe('POST /signin', () => {
     });
 
     const req = await request(app).post('/signin').send({
```
Comment (hynnot) [inline backend/src/tests/e2e/signin.test.ts:27]: This case it's not possible. According to our front, we need a test that is for username only and another for email only

**Review by hynnot — CHANGES_REQUESTED** [review]

File: backend/src/contexts/users/actions/SignInAction.test.ts (hunk 2 above)
Comment (Aumarh) [reply backend/src/contexts/users/actions/SignInAction.test.ts]: Done

File: backend/src/tests/e2e/signin.test.ts (hunk 3 above)
Comment (Aumarh) [reply backend/src/tests/e2e/signin.test.ts:27]: done

**Review by hynnot — APPROVED** [review]

**Pull request merged**